generators/php/model/src/union/UnionGenerator.ts

```diff
@@ -0,0 +1,1329 @@
+            safeName: "value",
+            unsafeName: "value"
+        };
+
+        return {
+            originalName: "value",
+            camelCase: safeUnsafeValue,
+            pascalCase: safeUnsafeValue,
+            snakeCase: safeUnsafeValue,
+            screamingSnakeCase: { safeName: "VALUE", unsafeName: "VALUE" }
+        };
+    }
+
+    private getDiscriminantField(): php.Field {
+        const includeGetter: boolean = this.context.shouldGenerateGetterMethods();
+        // TODO(ajgateno): Actually add the literals as a union rather than just string
+        return php.field({
+            name: this.unionTypeDeclaration.discriminant.wireValue,
+            type: php.Type.string(),
+            access: includeGetter ? "private" : "public",
+            readonly_: true
+        });
+    }
+
+    private getValueField(): php.Field {
+        const includeGetter: boolean = this.context.shouldGenerateGetterMethods();
+        const types: php.Type[] = [];
+
+        for (const variant of this.unionTypeDeclaration.types) {
+            types.push(this.getReturnType(variant));
+        }
+
+        types.push(php.Type.mixed());
+
+        return php.field({
+            // TODO(ajgateno): We'll want to disambiguate here if e.g. there's a "value" property
+            name: "value",
+            type: php.Type.union(types),
+            access: includeGetter ? "private" : "public",
+            readonly_: true
+        });
+    }
+
+    private toField({ property, inherited }: { property: ObjectProperty; inherited?: boolean }): php.Field {
+        const convertedType = this.context.phpTypeMapper.convert({ reference: property.valueType });
+        return php.field({
+            type: convertedType,
+            name: this.context.getPropertyName(property.name.name),
+            access: "public",
+            docs: property.docs,
+            attributes: this.context.phpAttributeMapper.convert({
+                type: convertedType,
+                property
+            }),
+            inherited
+        });
+    }
+
+    private unknownStaticConstructor(): php.Method {
+        const parameters: php.Parameter[] = [];
+
+        for (const property of this.unionTypeDeclaration.baseProperties) {
+            parameters.push(
+                php.parameter({
+                    name: this.context.getPropertyName(property.name.name),
+                    type: this.context.phpTypeMapper.convert({ reference: property.valueType })
+                })
+            );
+        }
+
+        parameters.push(
+            php.parameter({
+                name: "_unknown",
+                type: php.Type.mixed()
+            })
+        );
+
+        const body = php.codeblock((writer) => {
+            const constructorArgs: php.Map.Entry[] = [];
+
+            for (const property of this.unionTypeDeclaration.baseProperties) {
+                constructorArgs.push({
+                    key: php.codeblock(`'${property.name.wireValue}'`),
```

Comment:
We should set the properties based on the `this.context.getPropertyName` (not the wire representation).
```suggestion
                    key: php.codeblock(`'${this.context.getPropertyName(property.name)}'`),
```